core/azure-core/src/main/java/com/azure/core/polling/Poller.java

```diff
@@ -0,0 +1,217 @@
+     * @param pollerOptions configuration options for poller.
+     * @param pollOperation This is the operation to be called.
+     */
+    public Poller(PollerOptions pollerOptions,
+                  Function<PollResponse<T>, PollResponse<T>> pollOperation) {
+
+        this.pollerOptions = pollerOptions;
+        this.pollOperation = pollOperation;
+        fluxHandle = createFlux();
+        //autopolling  start here
+        setAutoPollingEnabled(true);
+    }
+
+    /**
+     * Constructor
+     * @param pollerOptions poller options
+     * @param pollOperation poller operation
+     * @param cancelOperation cancel operation
+     */
+    public Poller(PollerOptions pollerOptions,
+                  Function<PollResponse<T>, PollResponse<T>> pollOperation,
+                  Consumer<Poller> cancelOperation) {
+        this(pollerOptions, pollOperation);
+        this.cancelOperation = cancelOperation;
+    }
+
+    /*
+     * We will maintain single instance of fluxHandle for one poller.
+     */
+    private Flux<PollResponse<T>> createFlux() {
+        if (fluxHandle == null) {
+            fluxHandle = sendPollRequestWithDelay()
+                .flux()
+                .takeUntil(pollResponse -> needsPolling(pollResponse));
+        }
+        return fluxHandle;
+    }
+
+    private boolean needsPolling(PollResponse<T> pollResponse) {
+        return !isAutoPollingEnabled() && (pollResponse.getStatus() == PollResponse.OperationStatus.SUCCESSFULLY_COMPLETED
+            || pollResponse.getStatus() == PollResponse.OperationStatus.FAILED
+            || pollResponse.getStatus() == PollResponse.OperationStatus.USER_CANCELLED);
+    }
+
+    /**
+     * This will cancel polling from Azure Service if supported by service
+     * @throws UnsupportedOperationException when cancel operation is not suported.
+     */
+    public void cancelOperation() throws UnsupportedOperationException {
+        if (cancelOperation == null) {
+            throw new UnsupportedOperationException("Cancel operation is not supported on this service/resource.");
+        }
+        //We can not cancel an operation if it was never started
+        //or it is in its terminal state.
+        //Check cancelInitiated: to protect against multiple time call to cancel operation
+        if (cancelInitiated
+            || (pollResponse != null && pollResponse.getStatus() != PollResponse.OperationStatus.IN_PROGRESS)) {
+            return;
+        }
+        //Time to call cancel
+        cancelOperation.accept(this);
+        cancelInitiated = true;
+    }
+
+    /**
+     * This will poll and send PollResponse. If you had stopped polling erlier, we will enable polling again.
+     * @return Return poll response as Flux
+     */
+    public Flux<PollResponse<T>> poll() {
+        if (fluxDisposable != null && !fluxDisposable.isDisposed()) {
+            fluxDisposable.dispose();
+        }
+        return fluxHandle;
+    }
```

Comment:
I'm confused by the implementation here. Firstly, the conditional block appears to dispose something unexpectedly (to me). The comment in javadoc says it will enable polling again, which I 1) don't see happening in the implementation, and 2) don't agree that this is desired. This is also why I raised the question about differentiating between poll() and pollOnce(). Can you please consider this and give your vision here?

Reply:
1. Since I am changing implementation, the java doc became out of synch. I will keep it more insynch with code. 2.  I am thinking of poll should just return Flux and let client subscribe. Since we have autopolling, poll verb almost means client want to subscribe and receive all the events till it terminate.

Reply:
What happens in the case where a user has stopped auto polling and they call `poll()`? That would presumably not start auto-polling, they would just get a single poll response back and the flux would then close?

Reply:
just like you said : poll() is Manual polling mode. one response at a time. user has to call poll () in a loop and check status for completion.